node/router/src/lib.rs

```diff
@@ -177,13 +176,17 @@ impl<N: Network> Router<N> {
         if self.is_restricted(&peer_ip) {
             bail!("Dropping connection attempt to '{peer_ip}' (restricted)")
         }
+        // Ensure the node is not already connecting to this peer.
+        if !self.connecting_peers.lock().insert(peer_ip) {
```

Comment:
I was wondering if we should seperate this logic but so long as it's the last block in this function, it's ok. Splitting it out runs the risk of check and insert not operating on the same state.